### PR TITLE
Fix roundstarts failing when not enough thieves are available

### PR DESCRIPTION
**Pull request by bhenrich**
## About the PR

Fix a bug where a roundstart would fail if less users had the Thief antag enabled than the server expectes when trying to select thieves.

## Why / Balance

i see bug, ii fix, me happy

Closes Issue #24784

## Technical details

before picking the thieves using PickAntag(), determine the maximum number of thieves that can be picked from the available pool, without causing an ArgumentOutOfRangeException. then pick as many thieves as *safely* possible.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

none

**Changelog**

:cl: YuNii
- fix: Fix a bug where a roundstart would fail if the game failed to select enough thieves.

### CODE REVIEW

**Comment by TheShuEd** [comment]
#23445

**Comment by bhenrich** [comment]
I see my implementation more like a fix to keep this from happening until antag rework is finished

**Comment by Rainfey** [comment]
Go ahead, i dont anticipate pr #23445 being merged anytime soon - it will take a long time to get reviewed

File: Content.Server/GameTicking/Rules/ThiefRuleSystem.cs [hunk 1]
```diff
@@ -89,11 +89,42 @@ private void DoThiefStart(ThiefRuleComponent component)
...
+
+                }
+                else
+                {
+                    Log.Info("No potential thieves available.");
+                }
             }
+
+
         }
```
Comment (metalgearsloth) [inline Content.Server/GameTicking/Rules/ThiefRuleSystem.cs:128]: Is it possible to just share this code with nukies / traitors instead as every new antag we add is going to need to implement this.

Comment (VasilisThePikachu) [reply Content.Server/GameTicking/Rules/ThiefRuleSystem.cs:128]: If this is the case why not implement it in the already ongoing antag refactor?

Comment (deltanedas) [reply Content.Server/GameTicking/Rules/ThiefRuleSystem.cs:128]: it already is implemented there i think

Comment (bhenrich) [reply Content.Server/GameTicking/Rules/ThiefRuleSystem.cs:128]: uuh possible yes, but i'd have to learn a new system. right now this kinda bug only affects thief, the game knows how to handle any other antag in this situation. want me to do it anyways?

**Comment by TheShuEd** [comment]
Close as stale

if you come back to this, please let us know.

**Pull request closed without merging**